### ISSUE TITLE
Switching to use yt-dlp as youtube-dl seems to have stalled

### ISSUE THREAD

**Issue by lenaxia** [alexta69/metube#41]
a la https://github.com/ytdl-org/youtube-dl/issues/29753 it seems youtube-dl may have stalled

It seems there are some throttling issues in youtube-dl anyway (https://github.com/ytdl-org/youtube-dl/issues/29361) which I've been hitting a lot of.

So there's another drop in replacement for youtube-dl here: https://github.com/yt-dlp/yt-dlp and wanted to suggest switching to using that instead.

Cheers

**Comment by alexta69** [comment]
Thanks for the heads-up.
I've tried yt-dlp and it doesn't quite work as a drop-in, the download crashes, there must be something different there. I'll try to find time to debug this. If you have some time you're also welcome to try it out of course.. If I get to it first, I'll let you know.
Cheers!

**Comment by lenaxia** [comment]
I'm in the process of giving it a try as well, and likewise I also see the download crashing too. I'll see how far I can get to debug the crash.

**Comment by alexta69** [comment]
Have you tried the new version now, after I pushed the changes? it's supposed to work with yt-dlp now.
I did a little testing on it and it seems to work fine. Let me know if you're seeing problems with it.

**Comment by lenaxia** [comment]
Wow, that was fast. It works great. Looks like it defaults to webm format for download but thats something that can be managed on the user side. 

Below are the changes I made. The first one outputs mp4 if you select best quality because Plex doesn't seem to be able to read video descriptions from .webm files. And the second block downloads the thumbnail and write the title and video description into the mp4 file itself for plex to pick up. Right now I'm just mounting a copy of the file with edits as a docker volume.

```
if quality == 'best':
  self.format = 'bestvideo[ext=mp4]+bestaudio[ext=m4a]/best[ext=mp4]/best
```

```
ret = yt_dlp.YoutubeDL(params={                                                                                                                                                                                                                                                                                                                                            
  ...
  'writethumbnail': True,                                                                                                                                                                                                                                                                                                                                                         
  'postprocessors': [{                                                                                                                                                                                                                                                                                                                                                                 
  'key': 'FFmpegMetadata'                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       
    }],                                                                                                                                                                                                                                                                                                                                                                            
  ...
 }).download([self.info.url])
```

**Comment by alexta69** [comment]
Thanks! I liked the idea of preferring mp4, so I pushed the qualities change now. Note that in your example it seems, unless I'm missing something, that when you select a resolution it might still download the highest one. I pushed this, I hope it's correct (haven't tested, to be frank):
```
        elif quality in ('1440p', '1080p', '720p', '480p'):
            res = quality[:-1]
            self.format = f'bestvideo[height<={res}][ext=mp4]+bestaudio[ext=m4a]/best[height<={res}][ext=mp4]/best[height<={res}]'
```
I won't push the subtitle changes, because I feel the use case is too narrow and also AtomicParsley seems to be archeological -- it was last updated in 2005...

Thanks for your ideas!

**Comment by lenaxia** [comment]
I thought the same when I put that line of code in but I tried downloading a 480p version and it downloaded the right one so I left it as is. Can always strip out the end of the line if it does cause issues later.

And yeah, the subtitles and AtomicParsley may cause issues for others so totally get not inculding them. 

Happy to help! metube has been the best youtube-dl interface that I've found. It's simple and gets the job done well, so been super happy with it.
